### PR TITLE
Data Analyzer - Webhook v1

### PR DESCRIPTION
**Pull request by adrian-schnell**
WIP on https://github.com/defichain-api/masternode-health/issues/39

### CODE REVIEW

File: app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+                'type'      => 'block_height',
+                'value'     => $localBlockHeight,
+                'expected'  => $mainnetBlockHeight,
+                'explained' => sprintf('Possible chainsplit: node is %s blocks in front of main net', $diffBlockHeight),
+            ]);
+        }
+
+        $this->result->add([
+            'type'    => 'block_height',
+            'message' => sprintf('Node is has %s as latest block, main net is at block height %s', $localBlockHeight,
```
Comment (DerFuchs) [inline app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php:60]:
```suggestion
            'message' => sprintf('Node\'s block height is at $s, main net is currently at %s', $localBlockHeight,
```

File: app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+            $defidRunning = (bool)$this->getAttribute(ServerStatTypes::DEFID_RUNNING)->value;
+        } catch (AnalyzerException $e) {
+            return $this;
+        }
+
+        if (!$defidRunning) {
+            $this->critical->add([
+                'type'      => 'defid_running',
+                'value'     => $defidRunning,
+                'explained' => 'The defid seems not running fine. As long as it is not running, no blocks can be minted.',
```
Comment (DerFuchs) [inline app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php:116]:
```suggestion
                'explained' => 'The defid service seems to be either not running or not responding. It is possible that your node won\'t mint new blocks.',
```

File: app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+            $this->warnings->add([
+                'type'      => 'connection_count',
+                'value'     => $connectionCount,
+                'explained' => sprintf('At the moment there are only %s connections to the node. Maybe the node recently restartet. Otherwise it\'s recommended to add more connections.',
+                    $connectionCount),
+            ]);
+        }
+        $this->result->add([
+            'type'    => 'block_hash',
+            'message' => 'Block hashes of the node and the main net are equal',
```
Comment (DerFuchs) [inline app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php:146]: Block hash maybe wrong at this connection count analysis

Comment (DerFuchs) [reply app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php:146]: The whole block appears to be wrong.

File: app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+            $logSize = (float)$this->getAttribute(ServerStatTypes::LOGSIZE)->value;
+        } catch (AnalyzerException $e) {
+            return $this;
+        }
+
+        if ($logSize > 20) {
+            $this->warnings->add([
+                'type'      => 'logsize',
+                'value'     => $logSize,
+                'explained' => sprintf('The logfile seems to be quite big with %s MB.', $logSize),
```
Comment (DerFuchs) [inline app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php:165]:
```suggestion
                'explained' => sprintf('The logfile appears to be quite big with %s MB.', $logSize),
```

File: app/Api/v1/DataAnalyser/ServerStatAnalyzer.php [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        }
+
+        if ($currentRatio > 0.8) {
+            $this->warnings->add([
+                'type'      => 'hdd',
+                'value'     => $currentRatio,
+                'explained' => sprintf('%s percent of your HDD capacity used', $currentRatio),
+            ]);
+        }
+
```
Comment (DerFuchs) [inline app/Api/v1/DataAnalyser/ServerStatAnalyzer.php:73]: I suggest triggering a critial error when 99%+ (? maybe >95%) of HDD space is in use

File: app/Api/v1/DataAnalyser/ServerStatAnalyzer.php [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        if ($currentRatio > 0.8) {
+            $this->warnings->add([
+                'type'      => 'ram',
+                'value'     => $currentRatio,
+                'explained' => sprintf('%s percent of your RAM capacity used', $currentRatio),
+            ]);
+        }
+
```
Comment (DerFuchs) [inline app/Api/v1/DataAnalyser/ServerStatAnalyzer.php:102]: Same as HDD - a critical error should also be triggered

File: app/Client/CryptoidExplorerClient.php [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+use Str;
+
+class CryptoidExplorerClient
+{
+    protected Client $client;
+
+    public function __construct()
+    {
+        $this->client = new Client([
+            'base_uri' => 'https://chainz.cryptoid.info/dfi/api.dws',
```
Comment (DerFuchs) [inline app/Client/CryptoidExplorerClient.php:17]: Suggestion: Move that uri to a config file

File: app/Client/CryptoidExplorerClient.php [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        ]);
+    }
+
```
Comment (DerFuchs) [inline app/Client/CryptoidExplorerClient.php:20]: Another suggestion: Analyzer should not rely on a single source of data (cryptoid in this case), it should have at least one fallback to compensate possible downtimes. Maybe we should switch to DeFiChain Ocean ASAP and use that as primary souce.

File: app/Client/DefidVersion.php [hunk 9]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace App\Client;
+
+use GuzzleHttp\Client;
+use Str;
+
+class DefidVersion extends BaseGithubVersionClient
```
Comment (DerFuchs) [inline app/Client/DefidVersion.php:8]: This class seems not to be used anywhere

File: app/Client/CryptoidExplorerClient.php (hunk 8 above)
Comment (adrian-schnell) [reply app/Client/CryptoidExplorerClient.php:20]: let's do a 2nd source as separate issue. then we can compare them. are these ocean endpoints already running?

File: app/Client/DefidVersion.php (hunk 9 above)
Comment (adrian-schnell) [reply app/Client/DefidVersion.php:8]: valid point... this check was not implemented yet! will be in the next commit

File: app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php (hunk 3 above)
Comment (adrian-schnell) [reply app/Api/v1/DataAnalyser/NodeInfoAnalyzer.php:146]: fixed

**Pull request merged**